actors/miner/tests/miner_actor_test_partitions.rs

```diff
@@ -0,0 +1,1049 @@
+                termination_epoch,
+                &mut terminations,
+                SECTOR_SIZE,
+                QUANT_SPEC,
+            )
+            .unwrap();
+
+        // pop first termination
+        let (result, has_more) = partition.pop_early_terminations(&rt.store, 1).unwrap();
+
+        // expect first sector to be in early terminations
+        assert_bitfield_equals(&result.sectors[&termination_epoch], &[1]);
+
+        // expect more results
+        assert!(has_more);
+
+        // expect terminations to still contain 3 and 5
+        let queue = BitFieldQueue::new(&rt.store, &partition.early_terminated, QUANT_SPEC).unwrap();
+
+        // only early termination appears in bitfield queue
+        BitFieldQueueExpectation::default().add(termination_epoch, &[3, 5]).equals(&queue);
+
+        // pop the rest
+        let (result, has_more) = partition.pop_early_terminations(&rt.store, 5).unwrap();
+
+        // expect 3 and 5
+        assert_bitfield_equals(&result.sectors[&termination_epoch], &[3, 5]);
+
+        // expect no more results
+        assert!(!has_more);
+
+        // expect early terminations to be empty
+        let queue = BitFieldQueue::new(&rt.store, &partition.early_terminated, QUANT_SPEC).unwrap();
+        BitFieldQueueExpectation::default().equals(&queue);
+    }
+
+    #[test]
+    fn test_max_sectors() {
+        let (_, rt) = setup();
+        let mut partition = Partition::new(&rt.store).unwrap();
+
+        let proof_type = RegisteredSealProof::StackedDRG32GiBV1P1;
+        let _sector_size = proof_type.sector_size().unwrap();
+        let partition_sectors = proof_type.window_post_partitions_sector().unwrap();
+
+        let mut many_sectors = vec![SectorOnChainInfo::default(); partition_sectors as usize];
+        let mut ids = vec![0u64; partition_sectors as usize];
+        for (i, info) in many_sectors.iter_mut().enumerate() {
+            let id = (i as u64 + 1) << 50;
+            ids[i as usize] = id;
+            *info = test_sector(i as i64 + 1, id, 50, 60, 1000);
+        }
+        let sector_numbers = bitfield_from_slice(&ids);
+
+        let power = partition
+            .add_sectors(&rt.store, false, &many_sectors, SECTOR_SIZE, QUANT_SPEC)
+            .unwrap();
+        let expected_power = power_for_sectors(SECTOR_SIZE, &many_sectors);
+        assert_eq!(expected_power, power);
+
+        let empty = bitfield_from_slice(&[]);
+        assert_partition_state(
+            &rt.store,
+            &partition,
+            &many_sectors,
+            sector_numbers.clone(),
+            empty.clone(),
+            empty.clone(),
+            empty.clone(),
+            sector_numbers.clone(),
+        );
+
+        // Make sure we can still encode and decode.
+        let buf = RawBytes::serialize(&partition).expect("failed to marshal partition");
+        let _: Partition = buf.deserialize().unwrap();
+        assert_partition_state(
+            &rt.store,
+            &partition,
```

Comment:
To match the go test behavior you should assign the deserialized partition a value on 1037 and then use it here.